async.go

```diff
@@ -160,7 +161,34 @@ func getQueryResultWithRetriesForAsyncMode(
 			logger.WithContext(ctx).Errorf("failed to decode JSON. err: %v", err)
 			return respd, err
 		}
-		if respd.Code != queryInProgressAsyncCode {
+		if respd.Code == sessionExpiredCode {
+			// Update the session token in the header and retry
+			token, _, _ := sr.TokenAccessor.GetTokens()
+			if headers[headerAuthorizationKey] != fmt.Sprintf(headerSnowflakeToken, token) {
+				headers[headerAuthorizationKey] = fmt.Sprintf(headerSnowflakeToken, token)
+				logger.WithContext(ctx).Info("Session token has been updated.")
+				retry++
```

Comment:
Why do we bump this retry instead of `retryCountForSessionRenewal`?

Reply:
This is just to update the token that has been renewed by heartbeat already. `retryCountForSessionRenewal` is to check actual token renewal done in this function.